nilearn/reporting/html_report.py

```diff
@@ -0,0 +1,188 @@
+import io
+import base64
+import warnings
+from pathlib import Path
+from string import Template
+
+from nilearn.externals import tempita
+from nilearn.plotting import js_plotting_utils as plot_utils
+
+
+def _embed_img(display):
+    """
+    Parameters
+    ----------
+    display: obj
+        A matplotlib object to display
+
+    Returns
+    -------
+    embed : str
+        Binary image string
+    """
+    if display is None:  # no display, show single transparent pixel
+        data = ("iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAw" +
+                "CAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=")
+        return data
+
+    else:  # we were passed a matplotlib display
+        io_buffer = io.BytesIO()
+        display.savefig(io_buffer)
```

Comment:
could we embed the images as svg? (either base64 or simply url-encoded) it should look the same in the browser but if a user wants to save it and edit it it might be easier. it shouldn't require anything besides adding the format kwarg to savefig and replacing image/png with image/svg

Reply:
The (slight) problem I'm running into here is that the docstring was wrong -- I've since updated it !-- and I'm actually calling `.savefig` on a nilearn plotting object, which doesn't have access to the `format` kwarg. 

I don't want to save it out to disk, which is how the format is usually given here, so I'm not entirely sure how to make sure I'm creating an SVG. Any ideas ?

Reply:
Jerome helped me with this for my PR as well.

```python
import io
from matplotlib import pyplot as plt
buffer = io.StringIO()
plt.savefig(buffer, format='svg')
```

Reply:
> and I'm actually calling `.savefig` on a nilearn plotting object, which doesn't have access to the `format` kwarg.

you can access the figure through `nilearn_display.frame_axes.figure`

Reply:
and use the figure's `savefig`